### ISSUE TITLE
1071 specified key is too long

### ISSUE THREAD

**Issue by JGuiher** [NikolaGavric94/laravel-square#24]
I am very new to this kind of stuff, so sorry if I ask dumb questions.

I am following your tutorial on medium.com

I got to where I do the database migration, and it created some tables, but I got a 1071 specified key is too long error for the nikolag_taxes table.

It created the table in my database, but not the appropriate migration entry.

I have customers, customer_user, taxes and transactions for tables. Are there more missing?

I am on Laravel 5.7.27 so I have the default string length variable set at 191.

Thanks.

**Comment by NikolaGavric94** [comment]
Hello @JGuiher, could you post the full stacktrace? Thanks

**Comment by JGuiher** [comment]
I have no idea what a full stacktrace is. I did a search and maybe it is the full error thrown? If this doesn't help just tell me what I have to run to get what you need. Thanks.

![migration](https://user-images.githubusercontent.com/11841899/53404591-fd056900-3983-11e9-89c6-16d9cc18cf7e.JPG)

**Comment by NikolaGavric94** [comment]
Could you provide your db driver name and version too? Thanks

**Comment by JGuiher** [comment]
Is this it?

Server type: MySQL
Server version: 5.6.39-cll-lve - MySQL Community Server (GPL)
Protocol version: 10
User: cpses_osujrtosv8@localhost
Server charset: UTF-8 Unicode (utf8)
Web server
cpsrvd 11.70.0.51
Database client version: libmysql - 5.1.73
PHP extension: mysqliDocumentation curlDocumentation mbstringDocumentation
PHP version: 5.6.30

**Comment by NikolaGavric94** [comment]
I've detected the possible issue. Will test it out tonight and push a fix, stay tuned. Thanks for reporting @JGuiher

**Comment by JGuiher** [comment]
I updated all 6 of those files, still get the same exact error on the taxes table. Removed it and the rest of the migration ran perfect.

**Comment by NikolaGavric94** [comment]
I will need to run some tests then with your environment that you are using and get back to you. Sadly I'm cursed with really bad internet connection until Sunday so I will give my best to spin up some Docker containers and check it out, I'll get back to you soon then @JGuiher

**Comment by JGuiher** [comment]
![migration](https://user-images.githubusercontent.com/11841899/53449114-1ee40780-39e7-11e9-8702-0b28efe76b48.JPG)

Yes, sorry for the issue. I much appreciate the fast help.

**Comment by JGuiher** [comment]
I just tried an experiment, and it migrates properly if I give name and type a set length. 

I used 50.

I searched some and I think maybe it is that combining those as unique they are too long without a set max length?

**Comment by NikolaGavric94** [comment]
Can you check if your `engine` attribute inside `config/database.php`, `mysql` connection is `null`?

**Comment by JGuiher** [comment]
Yes it is.

Do you want me to change it to InnoDB?

It is in an answer I found here which made me try the length experiment.

https://stackoverflow.com/questions/42244541/laravel-migration-error-syntax-error-or-access-violation-1071-specified-key-wa

**Comment by NikolaGavric94** [comment]
You can change it and use it as a temporary solution until probably tomorrow, because at this time the connection gets really bad and i can't be of much help. Also, changing the length of all keys that make an index to `191` will do the trick, but since that will affect all users, I will need to find a better solution to include that rule only when really needed.

**Comment by NikolaGavric94** [comment]
@JGuiher Can you give me your environment info so I can replicate the issue? I will need your php version, correct laravel version (again) and mysql version (also again)

**Comment by JGuiher** [comment]
Apache Version | 2.4.33
-- | --
PHP Version | 5.6.36
MySQL Version | 5.6.39-cll-lve
Architecture | x86_64
Operating System | linux

**Comment by NikolaGavric94** [comment]
The library is installable for `php > 7.0.0` only. I've just tested against the same environment you have and the migrations passed successfully when I added bellow line to `AppServiceProvider:boot` method:

```suggestion
...
public function boot()
{
    Schema::defaultStringLength(191);
}
...
```
Laravel doesn't do this for you automatically because it can't assume which DB driver you are using and even less which version of it. Tested this against a fresh installment of laravel 5.7.27 and it works after adding the above line in the `AppServiceProvider:boot` method. I don't think this is a library issue, but I feel free to comment further here until we resolve your issue.
